### ISSUE TITLE
Evaluation with "model.eval()"

### ISSUE THREAD

**Issue by lslrh** [pengsongyou/openscene#20]
Thanks for your wonderful work!

When I run the 'distill.py' file, the evaluation performance can attain 41.0 miou with model.train(). But when I run the 'evaluate.py', the performance will drop to 25.0 miou with model.eval().

I don't know why the performance drop so significantly when I use model.eval() state?

**Comment by pengsongyou** [comment]
@lslrh, thanks for reaching out.

There are some layers disabled if you set to `model.eval()`, e.g. dropout, batch norm, etc. Also this might be due to optim.zero_grad(). You might want to take a look at [this](https://discuss.pytorch.org/t/performance-highly-degraded-when-eval-is-activated-in-the-test-phase/3323/4) and [this](https://medium.com/jun94-devpblog/pytorch-6-model-train-vs-model-eval-no-grad-hyperparameter-tuning-3812c216a3bd).

FYI, during inference, we actually use `model.eval()`, see [here](https://github.com/pengsongyou/openscene/blob/main/run/evaluate.py#L260-L261).

Best
Songyou

**Comment by lslrh** [comment]
> @lslrh, thanks for reaching out.
> 
> There are some layers disabled if you set to `model.eval()`, e.g. dropout, batch norm, etc. Also this might be due to optim.zero_grad(). You might want to take a look at [this](https://discuss.pytorch.org/t/performance-highly-degraded-when-eval-is-activated-in-the-test-phase/3323/4) and [this](https://medium.com/jun94-devpblog/pytorch-6-model-train-vs-model-eval-no-grad-hyperparameter-tuning-3812c216a3bd).
> 
> FYI, during inference, we actually use `model.eval()`, see [here](https://github.com/pengsongyou/openscene/blob/main/run/evaluate.py#L260-L261).
> 
> Best Songyou

Thanks. Why don't you use model.eval() in distill.py?

**Comment by pengsongyou** [comment]
@lslrh I just realize you said 

> But when I run the 'evaluate.py', the performance will drop to 25.0 miou with model.eval().
We indeed use model.eval() in `evaluate.py`, see [here](https://github.com/pengsongyou/openscene/blob/main/run/evaluate.py#LL261C9-L261C21), and you should get good results.

But for the validation in `distill.py`, indeed I just realized that I did not use `model.eval()` there. I will do check if that makes a big difference during training.
